### PR TITLE
Fixes #25648 - Delete filters as system admin

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Review by san7ket — DISMISSED** [review]
As a system admin, I was able to delete the filter on role successfully

File: app/controllers/filters_controller.rb [hunk 1]
```diff
@@ -69,7 +69,7 @@ def find_role
 
   def resource_base
     @resource_base ||= if @role.present?
-                         @role.filters.authorized(current_permission)
+                         Filter.authorized_as(User.current, current_permission)
```
Comment (ares) [inline app/controllers/filters_controller.rb]: as suggested on IRC, `Filter.authorized(current_permission)` would be nicer if it works

**Review by ares — CHANGES_REQUESTED** [review]

**Comment by xprazak2** [comment]
Changed, `Filter.authorized(current_permission)` worked for me.

**Review by ares — APPROVED** [review]

**Comment by ares** [comment]
Thanks @xprazak2, merging!

**Pull request merged**